src/deluge/io/midi/midi_device.h

```diff
@@ -152,16 +152,71 @@ class MIDIDeviceUSB : public MIDIDevice {
 	uint8_t portNumber;
 };
 
-class MIDIDeviceUSBHosted final : public MIDIDeviceUSB {
+class MIDIDeviceUSBHosted : public MIDIDeviceUSB {
 public:
 	MIDIDeviceUSBHosted() {}
 	void writeReferenceAttributesToFile();
 	void writeToFlash(uint8_t* memory);
 	char const* getDisplayName();
 
+	// Add new hooks to the below list.
```

Comment:
They can be moved, but I disagree as the current usage absolutely depends on vendor and product ID as a means of lookup, and these do not exist prior to MIDIDeviceUSBHosted. As well, current usage depends on the hostedMIDIDevices NamedThingsVector, which is not available to prior devices. Should this aspect change down the road it can be easily relocated.

Reply:
Ok fair enough